src/uct/ib/base/ece.h

```diff
@@ -0,0 +1,75 @@
+/**
+* Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef MLX5_ECE_H_
+#define MLX5_ECE_H_
+
+#include <stdint.h>
+#include <infiniband/verbs.h>
+#include <ucs/config/types.h>
+#include <ucs/debug/assert.h>
+
+
+enum  {
+    MLX5_ECE_DISABLED  = 0,
+    MLX5_ECE_VER_1     = 1,
+    MLX5_ECE_VER_SR    = 2,  /* selective repeat */
+    MLX5_ECE_VER_MAX   = MLX5_ECE_VER_SR,
+};
+
+
+typedef struct mlx5_ece_cfg {
+    ucs_on_off_auto_value_t ece_enable;
+    ucs_on_off_auto_value_t ece_sr;
+} mlx5_ece_cfg_t;
+
+
+union ece_t {
+    struct {
+        uint32_t sr  : 1;
+        uint32_t rsv : 27;
+        uint32_t ver : 4;
+    } field;
+
+    uint32_t val;
+};
+
+
+typedef struct mlx5_ece {
+    uint32_t ece_enable;
```

Comment:
Why the name is exactly  the same as line25? Are they the same thing?

Reply:
```mlx5_ece::ece_enable``` is the combination of ```user config option, mlx5_ece_cfg::ece_enable``` and ```device ece cap```.

Reply:
> 
@changchengx Then need different names for them.

Reply:
refine to be ```uint8_t enable``` and add comments about ```struct mxl5_ece``` and ```struct mlx5_ece_cfg```